cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -187,21 +189,33 @@ public String getPartition(PartitionKey key) {
         if (row == null) {
           break;
         }
-        if (filter != null && !validateFilter(filter, row)) {
+        PartitionKey key;
+        try {
+          key = parseRowKey(row.getRow(), partitioning);
+        } catch (IllegalArgumentException e) {
+          if (!ignoreInvalidRowsSilently) {
```

Comment:
the old data format can collide with scan range of the new format. If compatibility is turned on, then this is expected and we ignore it. If it is not turned on, then we expect no legacy rows, we still ignore these rows but we it is worth logging them.